pandas/core/arrays/sparse/array.py

```diff
@@ -927,29 +926,6 @@ def value_counts(self, dropna: bool = True) -> Series:
             index = keys
         return Series(counts, index=index)
 
-    def _quantile(self, qs: npt.NDArray[np.float64], interpolation: str):
-
-        if self._null_fill_value or self.sp_index.ngaps == 0:
-            # We can avoid densifying
```

Comment:
Was this optimization preventing the original sparse dtype from remaining? If not, I imagine this still being nice not to densify

Reply:
`SparseArray.__array__` already doesn't densify in this case.